sdk/typescript/src/module/entrypoint/register.ts

```diff
@@ -85,11 +100,17 @@ export class Register {
       })
 
       Object.values(enum_.values).forEach((value) => {
-        typeDef = typeDef.withEnumMember(value.name, {
+        const memberOpts: TypeDefWithEnumMemberOpts = {
           value: value.value,
           description: value.description,
           sourceMap: addSourceMap(value),
-        })
+        }
+
+        if (value.deprecated) {
```

Comment:
Same here, if `value.deprecated` is optional, you could directly assign it since `memberOpts.deprecated` type is `string | undefined`